spec/unit/functions/versioncmp_spec.rb

```diff
@@ -19,16 +19,44 @@ def versioncmp(*args)
   let(:type_parser) { Puppet::Pops::Types::TypeParser.singleton }
 
   it 'should raise an Error if there is less than 2 arguments' do
-    expect { versioncmp('a,b') }.to raise_error(/expects 2 arguments, got 1/)
+    expect { versioncmp('a,b') }.to raise_error(/expects between 2 and 3 arguments, got 1/)
   end
 
-  it 'should raise an Error if there is more than 2 arguments' do
-    expect { versioncmp('a,b','foo', 'bar') }.to raise_error(/expects 2 arguments, got 3/)
+  it 'should raise an Error if there is more than 3 arguments' do
+    expect { versioncmp('a,b','foo', false, 'bar') }.to raise_error(/expects between 2 and 3 arguments, got 4/)
   end
 
   it "should call Puppet::Util::Package.versioncmp (included in scope)" do
-    expect(Puppet::Util::Package).to receive(:versioncmp).with('1.2', '1.3').and_return(-1)
+    expect(Puppet::Util::Package).to receive(:versioncmp).with('1.2', '1.3', false).and_return(-1)
 
     expect(versioncmp('1.2', '1.3')).to eq(-1)
   end
+
+  context "when ignore_trailing_zeroes is true" do
+    it "should equate versions with dots and unnecessary zeroes" do
+      expect(versioncmp("10.1.0", "10.1", true)).to eq(0)
+    end
+
+    it "should equate versions with dots and iregular zeroes" do
+      expect(versioncmp("11.0.00", "11", true)).to eq(0)
+    end
+
+    it "should equate versions with dashes" do
+      expect(versioncmp("10.1-0", "10.1.0-0", true)).to eq(0)
+    end
+
+    it "should compare versions with dashes after normalization" do
+      expect(versioncmp("10.1-1", "10.1.0-0", true)).to eq(1)
+    end
+
+    it "should not normalize versions if zeros are not trailing" do
+      expect(versioncmp("1.1", "1.0.1", true)).to eq(1)
+    end
+  end
+
+  context "when ignore_trailing_zeroes is false" do
+    it "should not equate versions if zeros are not trailing" do
+      expect(versioncmp("1.1", "1.0.1")).to eq(1)
+    end
+  end
```

Comment:
I think there should also be a test for `11.0` and `11`, to verify that the original problem was solved.